src/components/ProductCard.test.jsx

```diff
@@ -110,18 +109,19 @@ describe('ProductCard', () => {
 
     mockProps.availableColors.forEach(async (colorOption) => {
       const colorDot = screen.getByTestId(`color-dot-${colorOption.color}`);
-      await waitFor(() => userEvent.click(colorDot));
```

Comment:
🟡 This caused an lint issue [Error: Avoid using side effects within 'waitFor' callback testing-library/no-wait-for-side-effects ]. `WaitFor` might execute its callback multiple times until the condition  is met. If there's a side effect within the callback, it'll be executed multiple times as well, which can lead to unexpected behavior in the  tests.  [https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-wait-for-side-effects.md]